dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs

```diff
@@ -82,118 +54,78 @@ public CultureInfo Culture
     /// <param name="errorDescription">Error description</param>
     /// <param name="exception">If available, the exception occurred</param>
     /// <returns>The current instance</returns>
-    public SKContext Fail(string errorDescription, Exception? exception = null)
-    {
-        this.ErrorOccurred = true;
-        this.LastErrorDescription = errorDescription;
-        this.LastException = exception;
-        return this;
-    }
+    public abstract void Fail(string errorDescription, Exception? exception = null);
 
     /// <summary>
     /// User variables
     /// </summary>
-    public ContextVariables Variables { get; }
+    public abstract ContextVariables Variables { get; }
 
     /// <summary>
-    /// Semantic memory
+    /// Read only skills collection
     /// </summary>
-    [Obsolete("Memory no longer passed through SKContext. Instead, initialize your skill class with the memory provider it needs.")]
-    public ISemanticTextMemory Memory
-    {
-        get => throw new InvalidOperationException(
-            "Memory no longer passed through SKContext. Instead, initialize your skill class with the memory provider it needs.");
-    }
+    public abstract IReadOnlySkillCollection Skills { get; }
 
     /// <summary>
-    /// Read only skills collection
+    /// App logger
     /// </summary>
-    public IReadOnlySkillCollection Skills { get; }
+    public abstract ILogger Logger { get; }
 
     /// <summary>
-    /// Access registered functions by skill + name. Not case sensitive.
-    /// The function might be native or semantic, it's up to the caller handling it.
+    /// Print the processed input, aka the current data after any processing occurred.
+    /// If an error occurred, prints the last exception message instead.
     /// </summary>
-    /// <param name="skillName">Skill name</param>
-    /// <param name="functionName">Function name</param>
-    /// <returns>Delegate to execute the function</returns>
-    public ISKFunction Func(string skillName, string functionName)
-    {
-        return this.Skills.GetFunction(skillName, functionName);
-    }
+    /// <returns>Processed input, aka result, or last exception message if any</returns>
+    public abstract override string ToString();
 
     /// <summary>
-    /// App logger
+    /// Create a clone of the current context, using the same kernel references (memory, skills, logger)
```

Comment:
Is this comment still accurate? Wasn't memory removed?

Reply:
Wow, no, is it ever not accurate. I've reduced to "Creates a clone of the current context".
I believe the Clone() method can also be obsoleted (at least on the public base class) once Kernel itself is responsible for instantiating and managing all Context objects. TBD whether Planner extensions will still require this.